Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -1009,6 +1222,13 @@ void SetPlaneDetectionEnabled(bool enabled)
             [sessionDelegate SetPlaneDetectionEnabled:enabled];
         }
 
+        bool TrySetMeshDetectorEnabled(const bool enabled)
+        {
+            meshDetectionEnabled = enabled;
```

Comment:
>meshDetectionEnabled [](http://example.com/codeflow?start=12&length=20)

do we need to cache this meshDetectionEnabled value twice? It seems like we could always maintain this value in the session delegate and grab it when needed from the session delegate.

I do not know the history behind why plane detection doesn't do this/but double caching this same value seems like itd be prone to getting out of sync/generating regressions

Reply:
Agreed, this was actually just a holdover of how it was handled in Android where we don't have the notion of a separate sessionDelegate, and the *DetectionEnabled property on SessionImpl probably should have been factored out.